lib/elixir/lib/calendar/duration.ex

```diff
@@ -5,10 +5,79 @@ defmodule Duration do
   A `Duration` struct represents a collection of time scale units,
   allowing for manipulation and calculation of durations.
 
-  Date and time scale units are represented as integers, allowing for both positive and negative values.
+  Date and time scale units are represented as integers, allowing for
+  both positive and negative values.
 
-  Microseconds are represented using a tuple `{microsecond, precision}`. This ensures compatibility with
-  other calendar types implementing time, such as `Time`, `DateTime`, and `NaiveDateTime`.
+  Microseconds are represented using a tuple `{microsecond, precision}`.
+  This ensures compatibility with other calendar types implementing time,
+  such as `Time`, `DateTime`, and `NaiveDateTime`.
+
+  ## Shifting
+
+  The most common use of durations in Elixir's standard library is to
+  "shift" the calendar types.
+
+      iex> Date.shift(~D[2016-01-03], month: 2)
+      ~D[2016-03-03]
+
+  In the example above, `Date.shift/2` automatically converts the units
+  into a Duration struct, although one can also be given directly:
+
+      iex> Date.shift(~D[2016-01-03], Duration.new!(month: 2))
+      ~D[2016-03-03]
+
+  It is important to notice that shifting is not an arithmetic operation.
+  For example, adding `date + 1 month + 1 month` does not yield the same
+  result as `date + 2 months`. Let's see an example:
+
+      iex> ~D[2016-01-31] |> Date.shift(month: 1) |> Date.shift(month: 1)
+      ~D[2016-03-29]
+
+      iex> ~D[2016-01-31] |> Date.shift(month: 2)
+      ~D[2016-03-31]
+
+  As you can see above, the results differ, which explains why operations
+  with durations are called "shift" rather than "add". This happens because,
+  once we add one month to `2016-01-31`, we get `2016-02-29`. Then adding
+  one extra month gives us `2016-03-29` instead of `2016-03-31`.
+
+  In particular, when applying durations to Calendar.ISO types:
+
+    * larger units (such as years and months) are applied before
+      smaller ones (such as weeks, hours, days, etc.)
```

Comment:
Easier for non-native speakers IMO:

```suggestion
      smaller ones (such as weeks, hours, days, and so on)
```